src/unix/sunos.c

```diff
@@ -692,13 +694,56 @@ void uv_free_cpu_info(uv_cpu_info_t* cpu_infos, int count) {
   uv__free(cpu_infos);
 }
 
+/*
+ * Inspired By:
+ * https://blogs.oracle.com/paulie/entry/retrieving_mac_address_in_solaris
+ * http://www.pauliesworld.org/project/getmac.c
+ */
+static int uv__set_phys_addr(uv_interface_address_t* address, struct ifaddrs* ent) {
+  struct sockaddr_dl* sa_addr;
+  int sockfd;
+  int i;
+  struct arpreq arpreq;
+
+  /* This appears to only work as root */
```

Comment:
The code after this comment inspects the results of `getifaddrs` to see if the `ifa_addr` field (which is copied into `phys_addr`) contains a non-zero byte. If it does contain a non-zero byte, then `getifaddrs` did successfully find the MAC address. I believe that this code path may only be successful if the code is running as root (which is what the comment is trying to indicate).